operator/main.py

```diff
@@ -464,22 +466,203 @@ def handle_modified(core_v1_client, obj):
 
     filename = os.path.join(MANIFESTS_DIR, "services.yaml")
     template(filename, namespace=NAMESPACE, volname=volname)
-    execute(KUBECTL_CMD, CREATE_CMD, "-f", filename)
+    lib_execute(KUBECTL_CMD, CREATE_CMD, "-f", filename)
     logging.info(logf("Deployed Service", volname=volname, manifest=filename))
 
+def delete_config_map(core_v1_client, obj):
+    """
+    Volinfo of new hosting Volume is generated and updated to ConfigMap
+    """
+    volname = obj["metadata"]["name"]
+    voltype = obj["spec"]["type"]
+    volume_id = obj["spec"]["volume_id"]
+
+    data = {
+        "namespace": NAMESPACE,
+        "kadalu_version": VERSION,
+        "volname": volname,
+        "volume_id": volume_id,
+        "type": voltype,
+        "bricks": [],
+        "options": obj["spec"].get("options", {})
+    }
+
+    # Add new entry in the existing config map
+    configmap_data = core_v1_client.read_namespaced_config_map(
+        KADALU_CONFIG_MAP, NAMESPACE)
+
+    # For each brick, add brick path and node id
+    bricks = obj["spec"]["storage"]
+    for idx, storage in enumerate(bricks):
+        data["bricks"].append({
+            "brick_path": "/bricks/%s/data/brick" % volname,
+            "kube_hostname": storage.get("node", ""),
+            "node": get_brick_hostname(volname, idx),
+            "node_id": storage["node_id"],
+            "host_brick_path": storage.get("path", ""),
+            "brick_device": storage.get("device", ""),
+            "pvc_name": storage.get("pvc", ""),
+            "brick_device_dir": get_brick_device_dir(storage),
+            "brick_index": idx
+        })
+
+    volinfo_file = "%s.info" % volname
+    configmap_data.data[volinfo_file] = ""
+
+    core_v1_client.patch_namespaced_config_map(
+        KADALU_CONFIG_MAP, NAMESPACE, configmap_data)
+    logging.info(logf("Updated configmap", name=KADALU_CONFIG_MAP,
+                      volname=volname))
+
+
+def delete_server_pods(obj):
```

Comment:
Accept `storage` instead of `obj` or along with `obj` so that it will be easier to get server pod/statefulset details.